File: sc-moyuan-backend/src/main/java/com/moyuan/entity/Rhyme.java
package com.moyuan.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("rhyme")
public class Rhyme {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String character;

    private String rhymeGroup;

    private String toneType;

    private String rhymeCategory;

    private Integer sortOrder;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
}

File: frontend/src/api/modules/rhyme.ts
import request from '@/utils/request'

export interface RhymeItem {
  id: number
  character: string
  rhymeGroup: string
  toneType: string
  rhymeCategory: string
  sortOrder: number
}

export function queryRhymeByCharacter(character: string) {
  return request.get<RhymeItem[]>('/rhyme/query', { params: { character } })
}

export function queryRhymeByGroup(group: string) {
  return request.get<RhymeItem[]>('/rhyme/group', { params: { group } })
}
